rewrite-gradle/src/main/java/org/openrewrite/gradle/RemoveDependency.java

```diff
@@ -78,6 +84,39 @@ protected TreeVisitor<?, ExecutionContext> getVisitor() {
             final MethodMatcher dependencyDsl = new MethodMatcher("DependencyHandlerSpec *(..)");
             final DependencyMatcher dependencyMatcher = requireNonNull(DependencyMatcher.build(groupId + ":" + artifactId).getValue());
 
+            @Override
+            public J visitJavaSourceFile(JavaSourceFile cu, ExecutionContext ctx) {
+                JavaSourceFile sourceFile = (JavaSourceFile) super.visitJavaSourceFile(cu, ctx);
+                if (sourceFile != cu) {
+                    Optional<GradleProject> maybeGp = sourceFile.getMarkers().findFirst(GradleProject.class);
+                    if (maybeGp.isPresent()) {
```

Comment:
Recipes are often written very defensively, with a long list of things to be checked, casted, null-safe'd before any change is made. To avoid ever-increasing indentation I find it more readable to return early.
```java
if(sourceFile != cu) {
     doStuff();
     if(otherCondition()) {
       doOtherStuff();
     }
}
return cu;
```
becomes the less-nested:
```java
if(sourceFile == cu) {
    return cu;
}
doStuff();
if(!otherCondition()) {
    return cu;
}
doOtherStuff();
return cu;
```

Reply:
Done via 807efaca908ce525a97466f52e91f7206a4d1039.